source/common/common/logger.h

```diff
@@ -358,4 +363,80 @@ template <Id id> class Loggable {
 
 // TODO(danielhochman): macros(s)/function(s) for logging structures that support iteration.
 
+/**
+ * ---------------------------------------------------------------
+ * Fancy logging macros
+ */
+
+using FancyMap = absl::flat_hash_map<std::string, std::shared_ptr<spdlog::logger>>;
+using FancyMapPtr = std::shared_ptr<FancyMap>;
+using SpdLoggerPtr = std::shared_ptr<spdlog::logger>;
+
+extern spdlog::level::level_enum kFancyDefaultLevel;
+
+extern absl::Mutex fancy_log_lock__;
+
+extern FancyMapPtr getFancyLogMap();
+
+extern void setFancyLogger(std::string key, spdlog::level::level_enum log_level);
```

Comment:
Why do you need extern for these functions?

If you simply want to declare and use them in the macros, would it make sense to create a fancy_logger.h and group these just there?

Reply:
Indeed I don't. Addressed now.